undefined reference to `tf2::BufferCore::_allFramesAsDot() const'

after #74 I get this error
/opt/ros/indigo/lib/libtf.so: undefined reference to `tf2::BufferCore::_allFramesAsDot() const'

to reproduce: create a workspace with the ros/velodyne repo and the ros/geometry_experimental repo, both of them checked out to the latest version.

I bisected on ros/geometry_experimental and I confirmed #74 is the culprit

Hmm, I just noticed you're having issues with allFramesAsDot that's different than the allFramesAsYAML which i'm reproducing. 

Some more info: 
First pass is producing a void argument symbol, rming devel and build produces double

```
buildfarm@7b33dfc8942c:/tmp/catkin_workspace$ nm /tmp/catkin_workspace/devel_isolated/tf2_ros/lib/libtf2_ros.so | grep YAML
                 U _ZNK3tf210BufferCore15allFramesAsYAMLEd
buildfarm@7b33dfc8942c:/tmp/catkin_workspace$ nm /tmp/libtf2_ros.so | grep YAML
                 U _ZNK3tf210BufferCore15allFramesAsYAMLEv
```

buffer server linking successfully: 

```
Linking CXX executable /tmp/catkin_workspace/devel_isolated/tf2_ros/lib/tf2_ros/buffer_server
/usr/bin/cmake -E cmake_link_script CMakeFiles/tf2_ros_buffer_server.dir/link.txt --verbose=1
/usr/bin/c++       CMakeFiles/tf2_ros_buffer_server.dir/src/buffer_server_main.cpp.o  -o /tmp/catkin_workspace/devel_isolated/tf2_ros/lib/tf2_ros/buffer_server -rdynamic /tmp/catkin_workspace/devel_isolated/tf2_ros/lib/libtf2_ros.so /opt/ros/indigo/lib/libactionlib.so /opt/ros/indigo/lib/libmessage_filters.so /opt/ros/indigo/lib/libroscpp.so -lboost_signals -lboost_filesystem /opt/ros/indigo/lib/librosconsole.so /opt/ros/indigo/lib/librosconsole_log4cxx.so /opt/ros/indigo/lib/librosconsole_backend_interface.so -llog4cxx -lboost_regex /opt/ros/indigo/lib/libxmlrpcpp.so /tmp/catkin_workspace/devel_isolated/tf2/lib/libtf2.so /opt/ros/indigo/lib/libroscpp_serialization.so /opt/ros/indigo/lib/librostime.so -lboost_date_time /opt/ros/indigo/lib/libcpp_common.so -lboost_system -lboost_thread -lpthread -lconsole_bridge -lboost_thread -lpthread -lconsole_bridge -Wl,-rpath,/tmp/catkin_workspace/devel_isolated/tf2_ros/lib:/tmp/catkin_workspace/devel_isolated/tf2/lib:/opt/ros/indigo/lib: 
```

Failed linking: 

```
Linking CXX executable /tmp/catkin_workspace/devel_isolated/tf2_ros/lib/tf2_ros/buffer_server
/usr/bin/cmake -E cmake_link_script CMakeFiles/tf2_ros_buffer_server.dir/link.txt --verbose=1
/usr/bin/c++       CMakeFiles/tf2_ros_buffer_server.dir/src/buffer_server_main.cpp.o  -o /tmp/catkin_workspace/devel_isolated/tf2_ros/lib/tf2_ros/buffer_server -rdynamic /tmp/catkin_workspace/devel_isolated/tf2_ros/lib/libtf2_ros.so /opt/ros/indigo/lib/libactionlib.so /opt/ros/indigo/lib/libmessage_filters.so /opt/ros/indigo/lib/libroscpp.so -lboost_signals -lboost_filesystem /opt/ros/indigo/lib/librosconsole.so /opt/ros/indigo/lib/librosconsole_log4cxx.so /opt/ros/indigo/lib/librosconsole_backend_interface.so -llog4cxx -lboost_regex /opt/ros/indigo/lib/libxmlrpcpp.so /tmp/catkin_workspace/install_isolated/lib/libtf2.so /opt/ros/indigo/lib/libroscpp_serialization.so /opt/ros/indigo/lib/librostime.so -lboost_date_time /opt/ros/indigo/lib/libcpp_common.so -lboost_system -lboost_thread -lpthread -lconsole_bridge -lboost_thread -lpthread -lconsole_bridge -Wl,-rpath,/tmp/catkin_workspace/devel_isolated/tf2_ros/lib:/tmp/catkin_workspace/install_isolated/lib:/opt/ros/indigo/lib: 
/tmp/catkin_workspace/devel_isolated/tf2_ros/lib/libtf2_ros.so: undefined reference to `tf2::BufferCore::allFramesAsYAML() const'
collect2: error: ld returned 1 exit status
```

--- a/tf2/include/tf2/buffer_core.h
+++ b/tf2/include/tf2/buffer_core.h
@@ -212,7 +212,11 @@ public:
   /** \brief A way to see what frames have been cached in yaml format
    * Useful for debugging tools
    */
-  std::string allFramesAsYAML(double current_time = 0) const;
+  std::string allFramesAsYAML(double current_time) const;
+
+  /** Backwards compatibility for #84
+  */
+  std::string allFramesAsYAML() const;
 
   /** \brief A way to see what frames have been cached
    * Useful for debugging

--- a/tf2/include/tf2/buffer_core.h
+++ b/tf2/include/tf2/buffer_core.h
@@ -297,7 +297,8 @@ public:
   /** \brief Backwards compatabilityA way to see what frames have been cached
    * Useful for debugging
    */
-  std::string _allFramesAsDot(double current_time = 0) const;
+  std::string _allFramesAsDot(double current_time) const;
+  std::string _allFramesAsDot() const;
 
   /** \brief Backwards compatabilityA way to see what frames are in a chain
    * Useful for debugging

--- a/tf2/src/buffer_core.cpp
+++ b/tf2/src/buffer_core.cpp
@@ -275,7 +275,7 @@ TimeCacheInterfacePtr BufferCore::allocateFrame(CompactFrameID cfid, bool is_sta
   } else {
     frames_[cfid] = TimeCacheInterfacePtr(new TimeCache(cache_time_));
   }
-  
+
   return frames_[cfid];
 }
 
@@ -1073,6 +1073,11 @@ std::string BufferCore::allFramesAsYAML(double current_time) const
   }
 
   return mstream.str();
+}
+
+std::string BufferCore::allFramesAsYAML() const
+{
+  return this->allFramesAsYAML(0.0);
 }
 
 TransformableCallbackHandle BufferCore::addTransformableCallback(const TransformableCallback& cb)

--- a/tf2/src/buffer_core.cpp
+++ b/tf2/src/buffer_core.cpp
@@ -1409,6 +1409,10 @@ std::string BufferCore::_allFramesAsDot(double current_time) const
   return mstream.str();
 }
 
+std::string BufferCore::_allFramesAsDot() const
+{
+  return _allFramesAsDot(0.0);
+}
 
 void BufferCore::_chainAsVector(const std::string & target_frame, ros::Time target_time, const std::string & source_frame, ros::Time source_time, const std::string& fixed_frame, std::vector<std::string>& output) const
 {

